client/db/bolt/db.go

```diff
@@ -459,6 +487,38 @@ func (db *boltDB) matchesUpdate(f bucketFunc) error {
 	return db.withBucket(matchesBucket, db.Update, f)
 }
 
+// UpdateWallet adds a wallet to the database.
+func (db *boltDB) UpdateWallet(wallet *dexdb.Wallet) error {
+	return db.walletsUpdate(func(wBkt *bbolt.Bucket) error {
+		return wBkt.Put(wallet.ID(), wallet.Encode())
```

Comment:
The wallet ID is just the byte-encoded asset ID. This means that there can only be one wallet per asset.

Reply:
One wallet at a time seems reasonable since you can restart with another wallet, but that cancels live orders.  Since you can just move funds between wallets instead, I think this is acceptable.